### PR TITLE
fix: false positive in `camelcase` with combined properties

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Fix https://github.com/eslint/eslint/issues/15572

```js
/*eslint camelcase: ["error", {"properties": "never", ignoreDestructuring: true}]*/

var { some_property } = obj; // OK
doSomething({ some_property }); // OK
```

#### Is there anything you'd like reviewers to focus on?

No

### CODE REVIEW

File: tests/lib/rules/camelcase.js
Comment (mdjermanovic) [inline tests/lib/rules/camelcase.js:428]: This test case might be confusing, as it doesn't have identifiers where `properties: "never"` applies. The same code is valid with the default `properties: "always"`, too - [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGNhbWVsY2FzZTogW1wiZXJyb3JcIiwgeyBwcm9wZXJ0aWVzOiBcImFsd2F5c1wiLCBpZ25vcmVEZXN0cnVjdHVyaW5nOiB0cnVlIH1dICovXG5cbmNvbnN0IHsgc29tZV9wcm9wZXJ0eSB9ID0gb2JqO1xuXG5jb25zdCBmb28gPSAoeyBzb21lX3Byb3BlcnR5IH0pID0+IHtcbiAgICBjb25zb2xlLmxvZyhzb21lX3Byb3BlcnR5KVxufTtcblxuZnVuY3Rpb24gYmFyKHsgc29tZV9wcm9wZXJ0eSB9KSB7XG4gICAgY29uc29sZS5sb2coc29tZV9wcm9wZXJ0eSlcbn07Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==).

`properties: "never"` applies only to identifiers used to create properties. Identifiers used to read properties are always allowed.

```js
/* eslint camelcase: ["error", { properties: "always", ignoreDestructuring: true }] */

x = obj.not_camelcase; // always allowed

obj.not_camelcase = x; // disallowed. can be allowed with properties: "never"

({ not_camelcase: x } = obj ); // always allowed

obj = { not_camelcase: x }; // // disallowed. can be allowed with properties: "never"


```

File: tests/lib/rules/camelcase.js
Comment (mdjermanovic) [inline tests/lib/rules/camelcase.js:438]:
```suggestion
            parserOptions: { ecmaVersion: 2022 }
```

This is a valid test case, so it shouldn't have `errors`.

File: lib/rules/camelcase.js
Comment (mdjermanovic) [inline lib/rules/camelcase.js:164]: This change looks good. Can you also update JSDoc description for this function since this doesn't fall under destructuring?

Comment (snitin315) [reply lib/rules/camelcase.js:164]: Updated 👍🏻

File: tests/lib/rules/camelcase.js
Comment (nzakas) [inline tests/lib/rules/camelcase.js:1458]: Can we add line numbers into this to make it clear which instance is being flagged?

File: tests/lib/rules/camelcase.js
Comment (nzakas) [inline tests/lib/rules/camelcase.js:1467]: Same here

**Review by nzakas — CHANGES_REQUESTED** [review]

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

Waiting for @nzakas's review before merging.

File: tests/lib/rules/camelcase.js
Comment (snitin315) [reply tests/lib/rules/camelcase.js]: Done.

**Review by nzakas — APPROVED** [review]
LGTM

**Pull request merged**
